### PR TITLE
Ability to suspend users (MongoDB)

### PR DESCRIPTION
**Pull request by Googlom**
#347

### CODE REVIEW

**Comment by or-else** [comment]
Please also update https://github.com/Googlom/chat/blob/suspend/server/db/mongodb/schema.md

**Comment by or-else** [comment]
One more thing: https://github.com/tinode/chat/commit/979eb31ebc18a39857ed8dbce2675f6d5186f288
I missed this rethinkdb change in my earlier commits.

File: server/db/mongodb/adapter.go [hunk 1]
```diff
@@ -1330,8 +1427,8 @@ func (a *adapter) TopicDelete(topic string, hard bool) error {
 	} else {
 		now := t.TimeNow()
 		_, err = a.db.Collection("topics").UpdateOne(a.ctx, filter, b.M{"$set": b.M{
-			"updatedat": now,
-			"deletedat": now,
+			"state":   t.StateDeleted,
```
Comment (aforge) [inline server/db/mongodb/adapter.go]: Do you want to update `updatedat` here as well?

Comment (Googlom) [reply server/db/mongodb/adapter.go]: https://github.com/tinode/chat/blob/1c74854863439fe5defa6d0c099be0d29cd4bf11/server/db/rethinkdb/adapter.go#L1244-L1249

`updatedat` is not updating in rethinkdb

Comment (or-else) [reply server/db/mongodb/adapter.go]: Let me check if it needs to be changed too. Maybe I made a mistake. I'll get back soon.

Comment (or-else) [reply server/db/mongodb/adapter.go]: Current code is correct. `updatedat` should NOT be updated on state change.

**Pull request merged**